Update README with information about live directives (#79)
balena-io-modules/livepush: README.md

will happen. After this the container will be restarted and
the `CMD` step will run again.

## Webpack-dev-server/nodemon/my custom monitor is faster!

We agree! Special case programs can be extremely fast, and
livepush does not stop you from using them from within a
container. To enable this, you can use live directives:

### dev-cmd-live

> A live cmd is the command that gets executed in the

Reviewer: ```suggestion
A live cmd, provided by the `dev-cmd-live` directive, is the command that gets executed in the
```
Might be worth being a bit more explicit here?

container when this container is running with livepush. It
overrides the existing `CMD` in the Dockerfile. This is

> achieved with a pre-process step on the Dockerfile.

Reviewer: ```suggestion
achieved with a preprocess step on the Dockerfile.
```

Original is fine, but you have used this variant in a couple of other places in the doc

```Dockerfile
FROM node

COPY package*.json ./

RUN npm install

#dev-cmd-live=webpack-dev-server

COPY ./src src

RUN npm run build

COPY nginx.conf .

CMD nginx
```

The above is a webpack based project which also describes

> how it should be ran with livepush. Livepush will preprocess

Reviewer: ```suggestion
how it should be run with livepush. Livepush will preprocess
```

this Dockerfile to be the equivalent of:

```Dockerfile
FROM node

COPY package*.json ./

RUN npm install

CMD webpack-dev-server

COPY ./src src

COPY nginx.conf .
```

Now when this container is executed, the main process will
be webpack-dev-server. Livepush will still copy in changed
source files. Livepush will optionally restart the container
based on a couple of rules:

- If the `COPY` step appears before the `#dev-cmd-live`
  directive, livepush will restart the container after
  copying in a file which is specified in that `COPY` step
- If the `COPY` step appears after the `#dev-cmd-live`
  directive, livepush will copy in the source file, but will
  not restart the container, and instead relies on the

>   command specified by the `#dev-cmd-live`

Reviewer: ```suggestion
  command specified by `#dev-cmd-live`
```

In this way, you can develop in a container, and have
blazing fast restart speeds via the program of your
choosing.

#### Multistage images

The behavior of a `#dev-cmd-live` in a multistage image is

> as follows;

Reviewer: ```suggestion
as follows:
```

- When the directive appears in the last stage, it is as
  described above.
- When the directive appears in any other stage, livepush
  will ignore any stages that follow.

For example:

```Dockerfile
FROM node AS build

COPY package*.json ./

RUN npm install

#dev-cmd-live=webpack-dev-server

COPY ./src src

RUN npm run build

FROM node

COPY --from=build /dist/app.js .

RUN nginx
```

will be preprocessed to:

```Dockerfile
FROM node AS build

COPY package*.json ./

RUN npm install

CMD webpack-dev-server

COPY ./src src
```

## I need other resources in my container when developing!

For this, you can use two other directives:

- `#dev-copy=`
- `#dev-run=`

These directives take exactly the same kind of arguments
as their Dockerfile counterparts, but are only executed
when the container is running as part of a livepush
process.

Using these, it is possible to bring in any extra
dependencies that are useful when developing. For example,

> one might choose to install webpack-dev-server using a

Reviewer: ```suggestion
you might choose to install webpack-dev-server using a
```

Totally optional but one seems a bit overly formal for this context.

> `dev-run` to avoid specifying it inside your package.json.

Reviewer: ```suggestion
`dev-run` directive to avoid specifying it inside your package.json.
```
